crafting_pieces.xml

```diff
@@ -668,7 +668,7 @@
       <Material id="Wood" count="1" />
     </Materials>
   </CraftingPiece>
-  <CraftingPiece id="crpg_practice_spear_t1_handle" name="{=Uhc0KIdz}Pine Spear Staff" tier="1" piece_type="Handle" mesh="spear_handle_1" length="149" weight="1.2" excluded_item_usage_features="long" CraftingCost="100" is_default="true">
+  <CraftingPiece id="crpg_practice_spear_t1_handle" name="{=Uhc0KIdz}Pine Spear Staff" tier="1" piece_type="Handle" mesh="spear_handle_1" length="160" weight="0.7" excluded_item_usage_features="long" CraftingCost="100" is_default="true">

```

Comment:
lengths need to be edited in scale factor in weapon.xml. Do not edit length in crafting piece . Then length in crafting piece is set by the 3d modeller to fin the 3d model. Editing thje length here wil resulte in gaps between parts because of a difference between physical model and 3d model. Since your branch is based on an outofdate , i highly recommend you follow first the recommandation i made in another  comment